nrf802154_sniffer/nrf802154_sniffer.py

```diff
@@ -356,11 +357,17 @@ def serial_reader(self, dev, channel, queue):
         """
         time.sleep(2)
 
+        timeout = time.time() + self.connection_open_timeout if self.connection_open_timeout else None
         while self.running.is_set():
             try:
                 self.serial = Serial(dev, timeout=1, exclusive=True)
                 break
             except Exception as e:
+                if timeout and time.time() > timeout:
+                    self.running.clear()
+                    raise Exception(
+                        "Could not open serial connection to sniffer before timeout of {} seconds".format(
+                            self.connection_open_timeout))
```

Comment:
This will kill the `serial_reader` thread, but it will still keep other threads alive, is it right? I think other threads should be killed as well after the timeout.

Reply:
Right, the othre threads were terminated as well while I was testing it